storage/innobase/handler/ha_innodb.cc

```diff
@@ -15651,6 +15675,10 @@ static int innodb_create_tablespace(handlerton *hton, THD *thd,
   trx_t *trx = check_trx_exists(thd);
   TrxInInnoDB trx_in_innodb(trx);
   trx_start_if_not_started(trx, true);
+  if (!innobase_is_ddse()) {
+    innobase_register_trx(hton, thd, trx);
+    mark_trx_rw(*thd);
```

Comment:
do you need similar change(mark_trx_rw) for innodb_drop_tablespace()

Reply:
`DROP TABLESPACE` transactions were already RW